network-api/networkapi/wagtailpages/pagemodels/buyersguide/homepage.py

```diff
@@ -392,25 +396,26 @@ def get_editorial_content_index(self):
         indexes = BuyersGuideEditorialContentIndexPage.objects.descendant_of(self)
         return indexes.first()
 
-    def get_hero_featured_article(self) -> Optional["BuyersGuideArticlePage"]:
+    def get_hero_featured_page(self) -> Optional[Union["BuyersGuideArticlePage", "BuyersGuideCampaignPage"]]:
         try:
-            return self.hero_featured_article.localized
+            return self.hero_featured_page.specific.localized
         except AttributeError:
-            # If no hero featured article is set (because `None` has no `localized`
+            # If no hero featured page is set (because `None` has no `localized`
             # attribute)
             return None
 
-    def get_hero_supporting_articles(self) -> list["BuyersGuideArticlePage"]:
-        articles = orderables.get_related_items(
-            self.hero_supporting_article_relations.all(),
-            "article",
+    def get_hero_supporting_pages(self) -> list[Union["BuyersGuideArticlePage", "BuyersGuideCampaignPage"]]:
+        supporting_pages = orderables.get_related_items(
+            self.hero_supporting_page_relations.all(),
+            "supporting_page",
         )
-        # FIXME: This implementation does return the localized version of each article.
-        #        But, it is inefficient. It would be better to pull all articles
-        #        for the correct locale at once. This would require the above returns
-        #        a queryset of the articles (rather than a list) and that we have an
-        #        efficient way of pulling all items for a given locale.
-        return [a.localized for a in articles]
+        # FIXME: This implementation returns the localized, subclassed version of each page.
+        #        But, it is inefficient. Both ".localized" and ".specific" are N+1 queries.
+        #        It would be better to pull all pages for the correct locale at once,
+        #        and applying ".specific()" at the end of the queryset.
+        #        This would require the above returns a queryset of the pages (rather than a list)
+        #        and that we have an efficient way of pulling all items for a given locale.
+        return [page.specific.localized for page in supporting_pages]
```

Comment:
What do you think about moving the `.specific()` call to the queryset, rather than the individual page? That should be more performant.

https://docs.wagtail.org/en/stable/reference/pages/queryset_reference.html#wagtail.query.PageQuerySet.specific